### ISSUE TITLE
Improve accessibility

### ISSUE THREAD

**Issue by basile-parent** [mdx-editor/editor#544]
If you want to ask for support or request features, [sponsor the project](https://github.com/sponsors/petyosi) and [contact me over email](mailto:petyo@mdxeditor.dev).

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] I have read the documentation and cannot find an answer.

**Describe the bug**
If found a few easy (and yet important) improvments on accessibility:
- We cannot put id on the textarea field (`div role="textbox"`) neither customizing its aria-label / aria-labelledby (set to `aria-label="editable markdown"`) so we cannot customize the accessible name

![image](https://github.com/user-attachments/assets/32b0b894-fa4b-4434-b521-47dd740d5783)

- All buttons have "radio" role which has no sense if they can be activated together (for exemple the bold, italic and underline buttons are independants, they shouldn't be specified by the "radio" role). Plus, the `button type="button"` is redundant. For me, the only button which should be identified as radio (and grouped into a proper fieldset) are:
    - The list buttons
    - The superscript / subscript buttons

![image](https://github.com/user-attachments/assets/048a74f0-b8a5-40c8-8a9f-f5c7879baafd)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Brave Version 1.50.120 | Chromium: 112.0.5615.121

**Comment by basile-parent** [comment]
PR in 2 parts : #552 & #553
